numba_dpex/core/types/usm_ndarray_type.py

```diff
@@ -24,42 +24,44 @@ def __init__(
         layout="C",
         dtype=None,
         usm_type="device",
-        device="unknown",
+        device=None,
         queue=None,
         readonly=False,
         name=None,
         aligned=True,
         addrspace=address_space.GLOBAL,
     ):
-        if not isinstance(device, str):
-            raise TypeError(
-                "The device keyword arg should be a str object specifying "
-                "a SYCL filter selector"
-            )
-
-        if not isinstance(queue, dpctl.SyclQueue) and queue is not None:
+        if queue and not isinstance(queue, types.misc.Omitted) and device:
             raise TypeError(
-                "The queue keyword arg should be a dpctl.SyclQueue object or None"
+                "numba_dpex.core.types.usm_ndarray_type.USMNdArray.__init__(): "
+                "`device` and `sycl_queue` are exclusive keywords, i.e. use one or other."
             )
 
         self.usm_type = usm_type
         self.addrspace = addrspace
 
-        if device == "unknown":
-            device = None
-
-        if queue is not None and device is not None:
-            raise TypeError(
-                "'queue' and 'device' keywords can not be both specified"
-            )
-
-        if queue is not None:
+        if queue and not isinstance(queue, types.misc.Omitted):
```

Comment:
If we have a non-None queue argument can it be anything but a `dpctl.SyclQueue`? Where can we end up with the queue being a `types.misc.Omitted`?

Reply:
> If we have a non-None queue argument can it be anything but a `dpctl.SyclQueue`? 

Then we fail on these tests `numba_dpex/tests/dpjit_tests/parfors/test_builtin_ops.py::test_built_in_operators2`

> Where can we end up with the queue being a `types.misc.Omitted`?

The type of queue becomes `types.misc.Omitted` when `dpnp.epmty()` is being called by dpnp functions internally.